langchain/document_loaders/blob_loaders/schema.py

```diff
@@ -0,0 +1,156 @@
+"""Schema for Blobs and Blob Loaders.
+
+The goal is to facilitate decoupling of content loading from content parsing code.
+
+In addition, content loading code should provide a lazy loading interface by default.
+"""
+import contextlib
+import mimetypes
+from abc import ABC, abstractmethod
+from io import BufferedReader, BytesIO
+from pathlib import PurePath
+from typing import Generator, Iterable, Optional, Union
+
+from pydantic import BaseModel
+
+PathLike = Union[str, PurePath]
+
+
+class Blob(BaseModel):
```

Comment:
Thoughts on adding a root_validator that asserts at least one of data or path is specified?

Reply:
Definitely good idea!